### PR TITLE
Update and patch for has many

### PR DESCRIPTION
**Pull request by Sergio-Mira**

### CODE REVIEW

**Comment by Sergio-Mira** [comment]
@davemswrve @cjkm Could you have a look? Cheers!

**Comment by davemswrve** [comment]
This is just the upgrade from 0.0.14 to 0.0.16 right?

Ignore that :)

**Comment by Sergio-Mira** [comment]
No, I added the patch to not trigger the hasMany relationship observers if the array doesn't change if ( a !== b) {
update
}

Also added a // SWRVE comment to anything that we've changed before

File: ember-model.js [hunk 1]
```diff
@@ -1371,18 +1429,22 @@ Ember.hasMany = function(type, options) {
...
-
-    if (arguments.length > 1) {
-      return existingArray.setObjects(newContentArray);
-    } else {
+      var key = options.key || propertyKey;
       return this.getHasMany(key, type, meta, this.container);
+    },
+    set: function(propertyKey, newContentArray, existingArray) {
+      //SWRVE-CHANGES
+      if (existingArray !== newContentArray) {
```
Comment (davemswrve) [inline ember-model.js]: Might it be better to use !Ember.isEquals here?

**Comment by davemswrve** [comment]
Yep I'm slow :)

File: ember-model.js (hunk 1 above)
Comment (robertclancy) [inline ember-model.js]: this is comparing arrays by reference, is this ok? or do we need to compare them based on the content of the array?

Comment (Sergio-Mira) [inline ember-model.js]: I've updated that to use isEquals

Comment (davemswrve) [inline ember-model.js]: I THINK comparison by reference is the idea as the observers are being fired even if the array is the same object.

Comment (Sergio-Mira) [inline ember-model.js]: I'll make a pull request against Ember to suggest they at least do the
check by reference (less expensive). That did work too for us for this case
as the collections are actually the same instance.

**Comment by Sergio-Mira** [comment]
The merge broke something. Will restore to latest copy and reapply our patches. Please do not merge for now.

**Pull request closed without merging**